### PR TITLE
Changed Responsepadding to only apply if fake header exists

### PR DESCRIPTION
**Pull request by mschulte-tsi**
Please Review

### CODE REVIEW

File: src/main/java/app/coronawarn/verification/controller/ExternalTestStateController.java [hunk 1]
```diff
@@ -125,4 +124,11 @@
       "Returning the test result for the registration token failed");
   }
 
+  private TestResult generateRetunTestResult(Integer testResult, String fake) {
```
Comment (mlaue-tech) [inline src/main/java/app/coronawarn/verification/controller/ExternalTestStateController.java:127]: "Retun" = Return ?

File: src/main/java/app/coronawarn/verification/controller/ExternalTanController.java [hunk 2]
```diff
@@ -143,4 +144,12 @@
       "VerificationAppSession not found for the registration token");
   }
 
+  private Tan generateRetunTan(String tan, String fake) {
```
Comment (mlaue-tech) [inline src/main/java/app/coronawarn/verification/controller/ExternalTanController.java:147]: "Retun" = Return ?

File: src/main/java/app/coronawarn/verification/model/Tan.java [hunk 3]
```diff
@@ -39,14 +40,15 @@
 @Data
 @AllArgsConstructor
 @RequiredArgsConstructor
+@NoArgsConstructor
 public class Tan {
 
   @NotNull
+  @NonNull
```
Comment (mlaue-tech) [inline src/main/java/app/coronawarn/verification/model/Tan.java:47]: Is Lombocks @NonNull neccessary?

File: src/main/java/app/coronawarn/verification/controller/ExternalTanController.java [hunk 4]
```diff
@@ -143,4 +144,12 @@
...
+    log.info(tan);
```
Comment (mlaue-tech) [inline src/main/java/app/coronawarn/verification/controller/ExternalTanController.java:148]: Are we allowed to log TANs?

File: src/main/java/app/coronawarn/verification/controller/ExternalTanController.java (hunk 2 above)
Comment (mschulte-tsi) [reply src/main/java/app/coronawarn/verification/controller/ExternalTanController.java:147]: Fixed typo

File: src/main/java/app/coronawarn/verification/controller/ExternalTestStateController.java (hunk 1 above)
Comment (mschulte-tsi) [reply src/main/java/app/coronawarn/verification/controller/ExternalTestStateController.java:127]: Fixed typo

File: src/main/java/app/coronawarn/verification/controller/ExternalTanController.java (hunk 4 above)
Comment (mschulte-tsi) [reply src/main/java/app/coronawarn/verification/controller/ExternalTanController.java:148]: Is Removed was for debuging

File: src/main/java/app/coronawarn/verification/model/Tan.java (hunk 3 above)
Comment (mschulte-tsi) [reply src/main/java/app/coronawarn/verification/model/Tan.java:47]: Yes ist is neccessary for lombock requiredargs

**Pull request merged**